mern-ecommerce/controllers/auth.js

```diff
@@ -80,3 +80,81 @@ exports.isAdmin = (req, res, next) => {
   }
   next();
 };
+
+exports.checkUser = (req, res, next) => {
+  let user = req.profile && req.auth && req.profile._id == req.auth._id;
+  if (!user) {
+    return res.status(403).jssosn({
+      error: 'Access denied',
+    });
+  }
+  next();
+};
+
+exports.forgotPassword = (req, res) => {
```

Comment:
The function `forgotPassword` is incomplete. It doesn't send an email to the user with the password reset instructions after generating the token. This could affect the functionality of the password reset feature.